src/state/CallViewModel/localMember/LocalMembership.ts

```diff
@@ -218,52 +216,23 @@ export const createLocalMembership$ = ({
     ),
   );
 
+  const localConnectionState$ = localConnection$.pipe(
+    switchMap((connection) => (connection ? connection.state$ : of(null))),
+  );
+
   // /**
   //  * Whether we are "fully" connected to the call. Accounts for both the
   //  * connection to the MatrixRTC session and the LiveKit publish connection.
   //  */
-  // // TODO use this in combination with the MemberState.
   const connected$ = scope.behavior(
     and$(
       homeserverConnected$,
-      localConnection$.pipe(
-        switchMap((c) =>
-          c
-            ? c.state$.pipe(map((state) => state.state === "ConnectedToLkRoom"))
-            : of(false),
-        ),
+      localConnectionState$.pipe(
+        map((state) => (state ? state.state === "ConnectedToLkRoom" : false)),
```

Comment:
When `state.state === "ConnectedToLkRoom"` should we then look at the inner `livekitConnectionState`? to look for `Disconnected` / `SignalReconnecting`?
```javascript
  | {
      state: "ConnectedToLkRoom";
      livekitConnectionState$: Observable<LivekitConenctionState>;
    }